g2_test.go

```diff
@@ -547,14 +549,53 @@ func TestG2HashToCurve(t *testing.T) {
 		},
 	} {
 		g := NewG2()
-		p0, err := g.HashToCurve(v.msg, domain)
+		p0, err := g.HashToCurve(sha256.New, v.msg, domain)
 		if err != nil {
 			t.Fatal("encode to point fails", i, err)
 		}
 		if !bytes.Equal(g.ToBytes(p0), v.expected) {
 			t.Fatal("encode to point fails x", i)
 		}
 	}
+	domain = []byte("QUUX-V01-CS02-with-expander")
```

Comment:
This looks like a 27 bytes = 216 bit DST, so this does not test the code path about the DST being longer than 255, does it?